From hjk@linutronix.de Fri Mar 14 03:28:52 2008
From: Jean-Samuel Chenard <jsamch@gmail.com>
Date: Fri, 14 Mar 2008 11:28:36 +0100
Subject: UIO: add pgprot_noncached() to UIO mmap code
To: Greg KH <greg@kroah.com>
Cc: Hans J Koch <hjk@linutronix.de>, linux-kernel@vger.kernel.org, Juergen Beisert <juergen127@kreuzholzen.de>, Jean-Samuel Chenard <jsamch@gmail.com>
Message-ID: <20080314112836.30f39afb@bluebox.local>

Mapping of physical memory in UIO needs pgprot_noncached() to ensure
that IO memory is not cached. Without pgprot_noncached(), it (accidentally)
works on x86 and arm, but fails on PPC.

Signed-off-by: Jean-Samuel Chenard <jsamch@gmail.com>
Signed-off-by: Hans J Koch <hjk@linutronix.de>
Cc: stable <stable@kernel.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/uio/uio.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/uio/uio.c
+++ b/drivers/uio/uio.c
@@ -470,6 +470,8 @@ static int uio_mmap_physical(struct vm_a
 
 	vma->vm_flags |= VM_IO | VM_RESERVED;
 
+	vma->vm_page_prot = pgprot_noncached(vma->vm_page_prot);
+
 	return remap_pfn_range(vma,
 			       vma->vm_start,
 			       idev->info->mem[mi].addr >> PAGE_SHIFT,
